lib/CodeGen/TargetLoweringObjectFileImpl.cpp

```diff
@@ -264,6 +265,7 @@ static SectionKind getELFKindForNamedSection(StringRef Name, SectionKind K) {
     return SectionKind::getThreadData();
 
   if (Name == ".tbss" ||
+      (TargetTriple.isAndroid() && Name == ".tcommon") || // LDC
```

Comment:
I've only found a single occurrence of `.tcommon`, as section for the `_tlsend` symbol. Is this line really only required because of that?

Reply:
Yes, it ensures that the final executable always has the `_tlsend` symbol after all emulated TLS symbols, believe it was a trick Walter used in the original version of this emulated TLS scheme for dmd with OS X, which I copied to ldc.

This reminds me though that I'll likely be taking this out once I finally get emulated TLS working with reading the ELF headers and ditching these delimiter symbols, so this will likely change before the final 1.12 release anyway.